### ISSUE TITLE
Evaluate performance on restored models: Mask dimensions error

### ISSUE THREAD

**Issue by pyvandenbussche** [Accenture/AmpliGraph#121]
### Description
the `evaluate_performance` function with filtering on fails on a pretrained model and its original data.
It throws an error `Number of mask dimensions must be specified, even if some dimensions` from tensorflow boolean mask. c.f. below.

Note that this succeed if we perform the evaluation unfiltered.

### Steps to Reproduce
You can reproduce this with original freebase data (fb15k-237) and best performing complex model trained on freebase fb15k-237.

```
    X = load_fb15k_237()

    # Restore the model
    model = restore_model(os.path.join("saved-models/", model_name))

    filter = np.concatenate((X['train'], X['valid']))

    # perform the evaluation using the learned model 
    ranks = evaluate_performance(X['test'],
                                 model=model,
                                 filter_triples=filter,
                                 use_default_protocol=True,  # corrupt subj and obj separately while evaluating
                                 verbose=True, strict=False)
```

**Comment by sumitpai** [comment]
Thanks for opening the issue. I just verified. This issue is mainly because you have not appended X['test'] in the filter, and generator gets a size zero list of filters for some triples (a case which is not handled in the code currently). 

Ideally in the SOTA computations of papers(ConvE and RotatE implementations) train+test+valid are added to filters (since all the triples are facts). But as you rightly pointed out, it need not be the case for other datasets where X_test may just be hand formulated test triples(which may not be facts). 

We will fix this on develop. Rather than appending the entire X_test, we would only add the sample that is being tested to the list of filters.

**Comment by pyvandenbussche** [comment]
Thanks for the comment. That's clear.
